lib/internal/streams/add-abort-signal.js

```diff
@@ -18,24 +24,28 @@ const validateAbortSignal = (signal, name) => {
   }
 };
 
-function isNodeStream(obj) {
-  return !!(obj && typeof obj.pipe === 'function');
-}
-
 module.exports.addAbortSignal = function addAbortSignal(signal, stream) {
   validateAbortSignal(signal, 'signal');
-  if (!isNodeStream(stream)) {
-    throw new ERR_INVALID_ARG_TYPE('stream', 'stream.Stream', stream);
+  if (!isNodeStream(stream) && !isWebStream(stream)) {
+    throw new ERR_INVALID_ARG_TYPE('stream', ['ReadableStream', 'WritableStream', 'Stream'], stream);
   }
   return module.exports.addAbortSignalNoValidate(signal, stream);
 };
+
 module.exports.addAbortSignalNoValidate = function(signal, stream) {
   if (typeof signal !== 'object' || !('aborted' in signal)) {
     return stream;
   }
-  const onAbort = () => {
-    stream.destroy(new AbortError(undefined, { cause: signal.reason }));
-  };
+  let onAbort;
+  if (isNodeStream(stream)) {
+    onAbort = () => {
+      stream.destroy(new AbortError(undefined, { cause: signal.reason }));
+    };
+  } else {
+    onAbort = () => {
+      stream[kControllerErrorFunction](new AbortError(undefined, { cause: signal.reason }));
+    };
+  }
```

Comment:
nit: we could keep `onAbort`  as a const
```suggestion
  const onAbort = isNodeStream(stream) ?
    () => {
      stream.destroy(new AbortError(undefined, { cause: signal.reason }));
    } :
    () => {
      stream[kControllerErrorFunction](new AbortError(undefined, { cause: signal.reason }));
    };
```